File: programmers/etc/level1/로또의_최고_순위와_최저_순위.py
def solution(lottos, win_nums):
    rank = {6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6}
    eq = len(set(lottos)&set(win_nums))
    zeros = lottos.count(0)
    return [rank[eq+zeros], rank[eq]]


lottos = [44, 1, 0, 0, 31, 25]
win_nums = [31, 10, 45, 1, 6, 19]
print(solution(lottos, win_nums)) # [3, 5]

lottos = [0, 0, 0, 0, 0, 0]
win_nums = [38, 19, 20, 40, 15, 25]
print(solution(lottos, win_nums)) # [1, 6]

lottos = [45, 4, 35, 20, 3, 9]
win_nums = [20, 9, 3, 45, 4, 35]
print(solution(lottos, win_nums)) # [1, 1]

File: Java/programmers/etc/level1/로또의최고순위와최저순위.java
package CodeTest.Java.programmers.etc.level1;

public class 로또의최고순위와최저순위 {
    public int[] solution(int[] lottos, int[] win_nums) {
        int eq = 0, zeros = 0;
        
        for (int lotto : lottos) {
            if (lotto == 0) zeros++;
            else {
                for (int win : win_nums) {
                    if (lotto == win) {
                        eq++;
                        break;
                    }
                }
            }
        }

        return new int[]{Math.min(7-(eq+zeros), 6), Math.min(7-eq, 6)};
    }
}
